File: Assignments/Assignment 1/app/src/main/java/com/example/assignment1/MainActivity.java
package com.example.assignment1;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void btnClick(View view) {
        Button btn = (Button)findViewById(R.id.michaelButton);
        if(btn.getText() == "Hello Michael!")
            btn.setText("Goodbye Michael!");
        else
            btn.setText("Hello Michael!");
    }
}

File: Assignments/Assignment 4/app/src/main/java/com/example/assignment4/MainActivity.kt
package com.example.assignment4

import android.content.Context
import android.content.Intent
import android.database.sqlite.SQLiteDatabase
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*;

class MainActivity : AppCompatActivity() {

    private lateinit var db: SQLiteDatabase

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        loadDatabase();
    }

    private fun loadDatabase() {
        db = openOrCreateDatabase("loginInformation", Context.MODE_PRIVATE, null)

        // Create the user table
        db.execSQL("DROP TABLE IF EXISTS users")
        db.execSQL("CREATE TABLE users (" +
                "username VARCHAR(32) PRIMARY KEY, " +
                "password VARCHAR(32))")

        // Insert valid login credentials
        db.execSQL("INSERT INTO users VALUES('admin', '1234')")
        db.execSQL("INSERT INTO users VALUES('admin2', '12345')")

    }

    fun btLoginClick(view: View) {
        var username = editTextUsername.text.toString()
        var password = editTextTextPassword.text.toString()

        println(username + " " + password)

        val query = "SELECT * FROM users WHERE username = " + username + " AND password = " + password

        val cursor = db.rawQuery(query, null)

        if(cursor.moveToFirst()) {
            val intent = Intent(this, MainActivity2::class.java)
            Toast.makeText(this, "Successful Login!", Toast.LENGTH_SHORT).show()
            startActivity(intent)
        }
        else {
            Toast.makeText(this, "Incorrect Credentials!", Toast.LENGTH_SHORT).show()
        }
    }


}